actions/middleware.go

```diff
@@ -101,3 +101,33 @@ func CheckAuthentication(appConfig *config.AppConfig, bux bux.ClientInterface, r
 	// Return an empty error message
 	return req, dictionary.ErrorMessage{}
 }
+
+// Request will write the request to the logs before and after calling the handler
+func Request(h httprouter.Handle, a *Action) httprouter.Handle {
+	return parameters.MakeHTTPRouterParsedReq(func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+		// Start the custom response writer
+		guid, _ := uuid.NewV4()
+		writer := &apirouter.APIResponseWriter{
+			IPAddress:      apirouter.GetClientIPAddress(req),
+			Method:         req.Method,
+			RequestID:      guid.String(),
+			ResponseWriter: w,
+			Status:         0, // future use with E-tags
+			URL:            req.URL.String(),
+			UserAgent:      req.UserAgent(),
+		}
+
+		// Start the log (timer)
+		start := time.Now()
+
+		// Fire the request
+		h(writer, req, ps)
+
+		// Complete the timer and final log
+		elapsed := time.Since(start)
+
+		if a.AppConfig.RequestLogging {
```

Comment:
Do we need `RequestLogging` flag if we log it as DEBUG?

Reply:
We can remove it If we want, It was still in config after refactoring so I used it

Reply:
Leave it as it is